### DISCUSSION TITLE
Replacing array with variable is not working using bash script

### DISCUSSION THREAD

**Discussion by dnyaneshwarjadhav** [mikefarah/yq#977] [Q&A] [answered]
**Describe the bug**
Replacing array with variable is not working using bash script.

Version of yq: 4.11.0
Operating system: windows
Installed via: chocolatey.

**Input Yaml**
Concise yaml document(s) (as simple as possible to show the bug, please keep it to 10 lines or less)
data1.yml:
```
_format_version: "2.1"

services:
- name: abc
- name: pqr

```
I am trying get all names into single array and then put back under some meta data tag.
For that

**Command**
var=yq e '.services[].name'  sample.yaml
//This is getting all the names into a variable 'var'

//And then trying to modify the same file
find sample.yaml -exec yq e '.a.info[] = [$var]' -i {} \;
```


**Actual behavior**

``File is not altering
```

**Expected behavior**

```yaml
_format_version: "2.1"

services:
- name: abc
- name: pqr

a
  info :
    - abc
    - pqr
```

**Additional context**
There is no error with script neither with commands of yq, just it dint worked for me.

**Comment by mikefarah** [comment, marked as answer]
Moved to discussion as you are asking for help with bash + yq

**Reply by dnyaneshwarjadhav** [reply]
So, any approach @mikefarah would you can help me here.

**Reply by mikefarah** [reply]
Did you actually check if you `var` assignment worked? from you've got there i don't think it did.

Regardless, you can just do

```
yq e -i '.a.info = [.services[].name]' sample.yaml
```

**Reply by dnyaneshwarjadhav** [reply]
@mikefarah yes assignment works for me, as that also you only helped me with bash commands syntax.
![image](https://user-images.githubusercontent.com/16952283/138277238-69a7de8b-e12e-45f3-b4d0-a33ad3db8033.png)

**Reply by dnyaneshwarjadhav** [reply]
> yq e -i '.a.info = [.services[].name]' sample.yaml

This instructions works for me, is there I can do with ordering of the newly placed element. It is always appending in the tail.
I wanted to provide position where it can go and append. is this possible ? thanks for ur 1 line of code it is made my 2 days struggle into 1 second job 👍

**Reply by mikefarah** [reply]
Glad I can help :)

The only way (I think) to put it first, is to construct a new map, (which will then have the new fields at the top) and then merge the old map into it, (which will put the old map fields in the bottom):


```
yq e '{"a": {"info": [.services[].name]}} *  .' examples/data1.yaml
```
